prime-router/src/main/kotlin/azure/BatchDeciderFunction.kt

```diff
@@ -40,46 +40,9 @@ class BatchDeciderFunction(private val workflowEngine: WorkflowEngine = Workflow
                     // any that should have batched in the last 60 seconds, get count of outstanding BATCH records
                     //  (how many actions with BATCH for receiver
                     .forEach { rec ->
-                        // Calculate how far to look back based on how often this receiver batches.
-                        val backstopTime = OffsetDateTime.now().minusMinutes(
-                            WorkflowEngine.getBatchLookbackMins(
-                                rec.timing?.numberPerDay ?: 1, NUM_BATCH_RETRIES
-                            )
-                        )
-                        // get the number of messages outstanding for this receiver
-                        val recordsToBatch = workflowEngine.db.fetchNumReportsNeedingBatch(
-                            rec.fullName, backstopTime, txn
-                        )
-                        var queueMessages = ceil((recordsToBatch.toDouble() / rec.timing!!.maxReportCount.toDouble()))
-                            .roundToInt()
-                        val logMessage = "$batchDecider found $recordsToBatch for ${rec.fullName}," +
-                            "max size ${rec.timing.maxReportCount}. Queueing $queueMessages messages to BATCH"
-                        if (recordsToBatch > 0) logger.info(logMessage)
-                        else logger.debug(logMessage)
-
-                        var isEmpty = false
-                        // if there are no records to send but the receiver is set on 'send when empty' check if
-                        //  a message should be added anyway
-                        if (rec.timing.whenEmpty.action == Receiver.EmptyOperation.SEND && queueMessages == 0) {
-
-                            val oneDayAgo = OffsetDateTime.now().minusDays(1)
-                            // lazy load this, may not need it if empty file is set to go every batch run
-                            val sentInLastDay: Boolean by lazy {
-                                workflowEngine.db.checkRecentlySent(
-                                    rec.organizationName,
-                                    rec.name,
-                                    oneDayAgo,
-                                    txn
-                                )
-                            }
-                            // determine if we need to send an 'empty' file. This is true if either we send an empty
-                            //  file every time the batch runs for this receiver of if we have not had a SEND
-                            //  action within the last 24 hours
-                            if (!rec.timing.whenEmpty.onlyOncePerDay || !sentInLastDay) {
-                                queueMessages = 1
-                                isEmpty = true
-                            }
-                        }
+                        val msgInfo = determineQueueMessageCount(rec, txn)
```

Comment:
I recommend you use something like 
val (queueMessage, isEmpty) = instead of the next two lines.